data-transfer-service/src/main/java/com/epam/pipeline/dts/sync/service/impl/DtsSynchronizationService.java

```diff
@@ -16,26 +16,244 @@
+import com.epam.pipeline.dts.transfer.model.pipeline.PipelineCredentials;
+import com.epam.pipeline.dts.transfer.repository.TaskRepository;
+import com.epam.pipeline.dts.transfer.service.TransferService;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.scheduling.annotation.EnableScheduling;
 import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.scheduling.support.CronSequenceGenerator;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
 
 @Service
-@RequiredArgsConstructor
+@Slf4j
 @EnableScheduling
 public class DtsSynchronizationService {
 
+    private static final String SCHEMA_DELIMITER = "://";
+
+    private final TaskRepository taskRepository;
+    private final TransferService transferService;
     private final PreferenceService preferenceService;
     private final ShutdownService shutdownService;
+    private final PipelineCredentials pipeCredentials;
+    private final Map<AutonomousSyncRule, AutonomousSyncCronDetails> activeSyncRules;
+    private final Map<AutonomousSyncRule, TransferTask> activeTransferTasks;
+    private final String defaultCronExpression;
+
+    @Autowired
+    public DtsSynchronizationService(final @Value("${dts.api.url}") String pipeApiUrl,
+                                     final @Value("${dts.api.token}") String pipeApiToken,
+                                     final @Value("${dts.autonomous.sync.cron}") String defaultCronExpression,
+                                     final TransferService autonomousTransferService,
+                                     final TaskRepository taskRepository,
+                                     final PreferenceService preferenceService,
+                                     final ShutdownService shutdownService) {
+        this.pipeCredentials = new PipelineCredentials(pipeApiUrl, pipeApiToken);
+        this.taskRepository = taskRepository;
+        this.transferService = autonomousTransferService;
+        this.shutdownService = shutdownService;
+        this.preferenceService = preferenceService;
+        this.activeSyncRules = new ConcurrentHashMap();
+        this.activeTransferTasks = new HashMap<>();
```

Comment:
As long as we have a method being executed by schedule then several threads can possibly access this map. I don't really think it may cause any issues here since we don't have any concurrent updates. But still I would use `ConcurrentHashMap` here just to be sure.